File: src/sql_jsc/postgres/AuthenticationState.rs
use super::sasl::SASL;

pub enum AuthenticationState {
    Pending,
    None,
    Ok,
    Sasl(SASL),
    Md5,
}

impl AuthenticationState {
    pub fn zero(&mut self) {
        // Assigning into *self drops the previous variant (and thus SASL's
        // Drop impl) automatically; no explicit deinit is needed.
        *self = AuthenticationState::None;
    }
}

File: src/sql_jsc/postgres/AuthenticationState.zig
pub const AuthenticationState = union(enum) {
    pending: void,
    none: void,
    ok: void,
    SASL: SASL,
    md5: void,

    pub fn zero(this: *AuthenticationState) void {
        switch (this.*) {
            .SASL => |*sasl| {
                sasl.deinit();
            },
            else => {},
        }
        this.* = .{ .none = {} };
    }
};

const SASL = @import("./SASL.zig");
